src/main/java/com/arpnetworking/tsdcore/model/CompositeUnit.java

```diff
@@ -0,0 +1,338 @@
+/**
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.tsdcore.model;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import net.sf.oval.constraint.NotNull;
+
+
+/**
+ * Composite Unit used in 2g file format.
+ *
+ * @author Ryan Ascheman (rascheman at groupon dot com)
+ */
+public class CompositeUnit {
+    /**
+     * Default constructor for Composite Unit.
+     */
+    public CompositeUnit() {}
+
+    /**
+     * Constructor for Composite Unit.
+     *
+     * @param scale Scale of the unit
+     * @param type Type fo the unit
+     */
+    public CompositeUnit(final Scale scale, final Type type) {
+        _type = type;
+        _scale = scale;
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj != null && obj instanceof CompositeUnit) {
+            final CompositeUnit compositeUnit = (CompositeUnit) obj;
+            return _type == compositeUnit._type && _scale == compositeUnit._scale;
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return _type.hashCode() ^ (_scale != null ? _scale.hashCode() : 1);
+    }
+
+    public void setType(final Type value) {
+        _type = value;
+    }
+
+    public void setScale(final Scale value) {
+        _scale = value;
+    }
+
+    /**
+     * Composite unit data type.
+     */
+    public enum Type {
```

Comment:
Where is this type used? There is already a type associated with each unit on the existing internal model and the file model does not as far as I can tell need this at all.